### ISSUE TITLE
SPARQL query "order by" on literal producing unexpected ordering

### ISSUE THREAD

**Issue by edmondchuc** [oxigraph/oxigraph#160]
I loaded the same data into both GraphDB and Oxigraph via Oxigraph.js using the `MemoryStore`.

I ran the following query:
```sparql
prefix skos: <http://www.w3.org/2004/02/skos/core#>
select *
where {
    ?concept a skos:Concept ;
             skos:prefLabel ?label .
}
order by ?label
```

The results were the same but the "order by" did not work as expected with Oxigraph's results.

Please see the diff of GraphDB and Oxigraph SPARQL results at https://www.diffchecker.com/3o7I9nj4.

Interestingly, it produced the same result as GraphDB when I tried with `order by ?concept`. The only difference I can think of is that one is that `?concept` is a `NamedNode` while `?label` is a `Literal`.

I will follow up with another test to load and run the same data and query in Oxigraph Server and see what the results are.

**Comment by Tpt** [comment]
I released as part of v0.3.0-beta.4 a change that orders all literals that were not already comparabled based on their "value" and if the values are equals on their datatype/language tag. I hope it will make the results more "consistent". Feel free to reopen this task if you disagree.

**Comment by edmondchuc** [comment]
Thanks @Tpt, I just tested it and the results feel more "correct" when viewed as a human now.
